mlflow/server/job_api.py

```diff
@@ -0,0 +1,69 @@
+"""
+Internal job APIs for UI invocation
+"""
+import os
+from typing import Any
+
+from fastapi import APIRouter, HTTPException, status as http_status
+from pydantic import BaseModel
+
+
+job_api_router = APIRouter(prefix="/ajax-api/3.0/job", tags=["Job"])
+
+
+_JOB_FUNCTION_ALLOW_LIST = [
+    # Add builtin job functions here
+]
+
+if extra_fun_allow_list_env := os.environ.get("_MLFLOW_JOB_FUNCTION_EXTRA_ALLOW_LIST"):
+    # this is for injecting allow listed job functions in test
+    _JOB_FUNCTION_ALLOW_LIST += extra_fun_allow_list_env.split(",")
+
+
+class QueryJobResponse(BaseModel):
+    """
+    Pydantic model for job query response.
+    """
+    status: str
+    result: Any
+
+
+@job_api_router.get("/query/{job_id}", response_model=QueryJobResponse)
```

Comment:
```suggestion
@job_api_router.get("/{job_id}", response_model=QueryJobResponse)
```

is there a reason `/query` is needed.